doc/specs/stdlib_math.md

```diff
@@ -91,6 +91,53 @@ program demo_clip_real
 end program demo_clip_real
 ```
 
+### `gcd` function
+
+#### Description
+
+Returns the greatest common divisor of two integers.
+
+#### Syntax
+
+`res = [[stdlib_math(module):gcd(interface)]] (a, b)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument(s)
+
+`a`: An integer.
+`b`: Another integer.
+
+Note: All arguments must be integers of the same `kind`.
+
+#### Output value or Result value
+
+The output is a scalar integer of the same `kind` as to that of the arguments.
+
+#### Examples
+
+##### Example 1:
+
+```fortran
+program demo_gcd
+  use stdlib_math, only: gcd
+  use stdlib_kinds, only: int32
+  implicit none
+  integer(int32) :: a, b, c
+  
+  a = 48_int32
+  b = 18_int32
```

Comment:
```suggestion
  a = 48
  b = 18
```

Kind suffix not necessary, compiler will cast it automatically.

Reply:
I took clip as a reference. Probably not the best choice.

Reply:
I would rather strip all the kinds for the example for simplicity. Or is it recommended to use explicit kinds in the examples? Still use a and b as input variables or simplify to use literals in the call?

Reply:
I suggest keeping `a` and `b` as declared variables. I would also just use default `integer` without `int32`. Makes the example simpler, and I can't think of a situation where it's problematic (i.e. default `integer` is `int32` on all platforms that I know of).

Reply:
Ok, I thought there was some underlying theme in the fact that clip uses int32 explicitly and didn't want to break them right next to it.